### ISSUE TITLE
Unimplemented: opx.printer

### ISSUE THREAD

**Issue by tomsci** [inseven/opolua#340]
Eg https://archive.org/download/3-libjune-05/3LIBJUNE05.iso/library%2Fepocmisc%2Fatmos5%2Fatmos522.SIS 

Print operations are unlikely to ever be supported, unless someone wants to send me a working psion5 compatible printer to test with...

**Comment by jbmorley** [comment]
I wonder if it would help to stub this out?

**Comment by kapfab** [comment]
Were there many third-party OPL apps that supported printing? A few banking apps maybe?

**Comment by jbmorley** [comment]
Good question. I've no clue at this point. I'm currently stuck in the middle of a refactor of the software index importer, but as soon as I've done that I'm going to start pulling out information about which OPXs are used, etc.

**Comment by tomsci** [comment]
> Were there many third-party OPL apps that supported printing? A few banking apps maybe?

I think this is the first one I've come across, and printers weren't that common hooked up to Psions in the first place I don't think.

> I wonder if it would help to stub this out?

I stubbed out `ResetPrinting()` which was enough to get atmos to at least not crash when showing the menu.

**Comment by jbmorley** [comment]
You know, if we've got no printing to reset, it sounds a lot like you've fully implemented `ResetPrinting()`.

**Comment by kapfab** [comment]
> I think this is the first one I've come across, and printers weren't that common hooked up to Psions in the first place I don't think.
Well, Psions came with Canon Bubblejet, HP Deskjet and LaserJet drivers and there were third-party PCL and Postscript drivers so that made quite a lot of printers compatible. And later came PDFPrinter to print as PDF. And printing was also possible via PC (PsiWin / plptools).
However, I can't remember me printing with third-party apps, only Psion's stock apps.

**Comment by jbmorley** [comment]
PDFPrinter would be interesting as on iOS we could actually use that to bring up a print screen.

**Comment by kapfab** [comment]
It was a specific printer driver developed by the same author as PDF+, he reimplemented the whole PDF generation stack, translating the printing instructions into a PDF.
The printing stack in Opolua (if you ever decide to implement it) should also be independent from any driver and translate the printer.opx configuration and drawing calls into a PDF.

**Comment by jbmorley** [comment]
Yeah. That makes sense. I imagine we'd have to do something from scratch if we went that way. Oh well, not for today, that's for sure. 🙃
